static/src/javascripts/projects/common/modules/commercial/commercial-features.ts

```diff
@@ -82,6 +84,28 @@ class CommercialFeatures {
 			!isHosted &&
 			!newRecipeDesign;
 
+		if (isArticle && !this.articleBodyAdverts) {
+			const articleBodyAdvertsFalseIfTrue: Record<string, boolean> = {
+				forceAdFree,
+				isAdFreeUser: isAdFreeUser(),
+				sensitiveContent,
+				isMinuteArticle,
+				isLiveBlog,
+				isHosted,
+				newRecipeDesign: !!newRecipeDesign,
+			};
```

Comment:
I like how clear `articleBodyAdvertsFalseIfTrue` and `IfFalse` are but even though these are very close to `articleBodyAdverts`, I can see it being a bit problematic that they're defined separately. It doesn't seem unlikely that someone in the future could come along and change the definition of `articleBodyAdverts` but neglect to update `articleBodyAdvertsFalseIfTrue` or `IfFalse` accordingly.
Could we unify them to avoid this?

Reply:
@zekehuntergreen The alternative I think is to keep `..IfTrue` with a different name, and rather than check for false it needs to have a special case for `switches.commercial` and `externalAdvertising`, which is a bit less elegant, and seems just as unlikely to get neglected from updating?

Reply:
I think I could clear it up by moving it to it's own function actually

Reply:
OR.. were you thinking to derive `articleBodyAdverts` from `ifTrue` and `ifFalse`?🤔

Reply:
Sorry I think my initial comment wasn't very clear. I was thinking that if we were to define a variable with the list of conditions necessary to show ads in articles, then use that variable both for the log and to define `this.articleBodyAdverts`, then we could make it almost impossible for a dev to make the mistake of letting the two (logging and definition of `articleBodyAdverts`) diverge.
I was thinking something along these lines for article body adverts and something similar for dfpAdvertising:
```
const articleBodyAdverts = {
	adFree: !this.adFree,
	isMinuteArticle: !isMinuteArticle,
	isArticle: isArticle,
	isLiveBlog: !isLiveBlog,
	isHosted: !isHosted,
	newRecipeDesign: !newRecipeDesign,
};

for (const [key, value] of Object.entries(articleBodyAdverts)) {
	if (!value) {
		log(
			'commercial',
			`Article adverts not shown because ${key} is ${value.toString()}`,
		);
	}
}

this.articleBodyAdverts =
	Object.values(articleBodyAdverts).every(Boolean);
```
Happy to be challenged on this though if you think it's unnecessary or overly cautious!